### PR TITLE
feat: Variable width for property pane size and side by side editor

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR adds adaptive width to property pane and side by side editor if screen size if bigger than 1440.
https://www.notion.so/appsmith/IDE-Solution-behaviour-8607a3b4df254970a77bd077290d3690#05be7aee530e4669ab6c79bb34ffadc7

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/30086

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced flexibility in property pane width settings, allowing both number and string inputs.
	- Added side-by-side mode for improved layout management.
	- Implemented new constants for default sidebar, editor pane, and design base widths.
	- Enhanced editor pane with transition effects and dynamic width adjustments based on editor mode.
	- Added UI control for canvas scaling, allowing users to adjust the canvas size.
- **Enhancements**
	- Updated `PropertyPaneWrapper` to use new hooks for dynamic width management.
	- Improved device detection with new functions for better layout responsiveness.
- **Refactor**
	- Renamed `useEditorPaneWidth` hook to `useIDEWidths` for clearer purpose and functionality.
- **Style**
	- Added CSS rule to limit maximum width of segmented header elements for better layout control.

### CODE REVIEW

File: app/client/src/constants/AppConstants.ts [hunk 1]
```diff
@@ -44,6 +46,7 @@ export const MOBILE_MAX_WIDTH = 767;
 export const TABLET_MIN_WIDTH = 768;
 export const TABLET_MAX_WIDTH = 991;
 export const DESKTOP_MIN_WIDTH = 992;
+export const DESIGN_BASE_WIDTH = 1440;
```
Comment (hetunandu) [inline app/client/src/constants/AppConstants.ts:49]: Design base width feels ambiguous. Can we call it App or Editor instead?
```suggestion
export const EDITOR_BASE_WIDTH = 1440;
```

Comment (hetunandu) [reply app/client/src/constants/AppConstants.ts:49]: Or even desktop

**Review by hetunandu — CHANGES_REQUESTED** [review]
Just a variable name change required

File: app/client/src/constants/AppConstants.ts [hunk 2]
```diff
@@ -12,6 +12,9 @@ export const CANVAS_BACKGROUND_COLOR = "#FFFFFF";
 export const DEFAULT_ENTITY_EXPLORER_WIDTH = 256;
 export const DEFAULT_PROPERTY_PANE_WIDTH = 288;
 export const APP_SETTINGS_PANE_WIDTH = 525;
+export const DEFAULT_APP_SIDEBAR_WIDTH = 50;
+export const DEFAULT_EDITOR_PANE_WIDTH = 255;
+export const DEFAULT_PP_LARGE_WIDTH = window.innerWidth * 0.404 - 255; // 40.4% - 255px
```
Comment (marks0351) [inline app/client/src/constants/AppConstants.ts:17]: can we add comments explaining the rationale here.

File: app/client/src/pages/Editor/Canvas.tsx [hunk 3]
```diff
@@ -26,19 +26,22 @@ interface CanvasProps {
...
 const Wrapper = styled.section<{
   background: string;
   width: number;
   $enableMainCanvasResizer: boolean;
+  $scale?: number;
 }>`
   background: ${({ background }) => background};
   width: ${({ $enableMainCanvasResizer, width }) =>
     $enableMainCanvasResizer ? `100%` : `${width}px`};
+  transform: ${({ $scale }) => ($scale ? `scale(${$scale})` : "")};
```
Comment (marks0351) [inline app/client/src/pages/Editor/Canvas.tsx:41]: pls ensure Drag and drop works properly when scaled.

Comment (marks0351) [reply app/client/src/pages/Editor/Canvas.tsx:41]: tested the DnD and I see it not working in all scenarios, coz this scale factor needs to be used for the canvas on which the DnD illusion is rendered. pls check StickyCanvasArena for more details. I remember @hetunandu implemented a similar scale factor based on a previous redesign project which involved creating a react state for it.

Also with this scaling I assume we might need to implement horizontal auto scroll while dragging and dropping. this itself is not a deal breaker for the first cut though.

File: app/client/src/pages/Editor/WidgetsEditor/MainContainerWrapper.tsx [hunk 4]
```diff
@@ -253,6 +255,41 @@ function MainContainerWrapper(props: MainCanvasWrapperProps) {
           </div>
         )}
         {node}
+        <Flex
```
Comment (marks0351) [inline app/client/src/pages/Editor/WidgetsEditor/MainContainerWrapper.tsx:258]: I would recommend moving this new block into a seperate component.

Comment (marks0351) [inline app/client/src/pages/Editor/WidgetsEditor/MainContainerWrapper.tsx:258]: I would recommend moving this new block into a seperate component.

**Review by marks0351 — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/WidgetsEditor/MainContainerWrapper.tsx (hunk 4 above)
Comment (albinAppsmith) [reply app/client/src/pages/Editor/WidgetsEditor/MainContainerWrapper.tsx:258]: This part is just poc for now, will account for your comments if we go ahead with this changes.

**Pull request merged**